extensions/amp-form/0.1/form-validators.js

```diff
@@ -242,6 +293,12 @@ export class AbstractCustomValidator extends FormValidator {
     if (!input.id) {
       return null;
     }
+    // <textarea> only supports `pattern` matching. But, it's implemented via
+    // setCustomValidity(), which results in the 'customError' validity state.
+    if (input.tagName === 'TEXTAREA') {
+      devAssert(invalidType === 'customError');
```

Comment:
The `customError` type is also used for form-verifiers. To harden the textarea fake-patternMismatch feature and make it less likely to conflict with a form-verifier we could key on a special value for `validationMessage` like `__amp_patternMismatch`

Reply:
If the publisher doesn't provide custom validation, would this result in displaying `__amp_patternMismatch` as the validation message to the user?

Looks like form-verifiers.js checks element validity first before calling `setCustomValidity()`, but I suppose we need to worry about the converse -- form validator overriding a verification error. I'll change `checkInputValidity()` to avoid this.

Any other edge cases you can think of?

Reply:
Oh good point, it would get appear when it's reported by the `form-validators.DefaultValidator#report` method. We could make that method aware of the patternMismatch implementation for textarea elements.

No other edge cases come to mind now, but I can let you know.

Reply:
Cool, thanks. Submitting.